src/ray/raylet/scheduling_resources.cc

```diff
@@ -505,12 +669,42 @@ const ResourceSet &SchedulingResources::GetLoadResources() const {
 
 // Return specified resources back to SchedulingResources.
 bool SchedulingResources::Release(const ResourceSet &resources) {
-  return resources_available_.AddResourcesStrict(resources);
+  bool release_status = resources_available_.AddResourcesCapacityConstrained(resources, resources_total_);
```

Comment:
We need to constrain `Release` now because resources may get removed while they are acquired by a task, and a subsequent release would incorrectly re-add those resources to `resources_available_`. 

I have removed `AddResourcesStrict`.